internal/host/hostcommands/install_cmd.go

```diff
@@ -287,7 +292,48 @@ func constructHostInstallerArgs(cluster *common.Cluster, host *models.Host, inve
 		}
 	}
 
-	if !hasStaticNetwork {
+	// Check if Manufacturer is IBM/S390 and ProductName is not "KVM/Linux" (the case for z/VM and LPAR).
+	// If this is the case than we need to extract the necessary z/VM kargs and append them.
+	if inventory != nil &&
+		inventory.SystemVendor != nil {
+		log.Debugf("Check for SystemVendor and ProductName: %s:%s\n", inventory.SystemVendor.Manufacturer, inventory.SystemVendor.ProductName)
+	} else {
+		log.Debug("Check for SystemVendor and ProductName: <nil>:<nil>\n")
+	}
+
+	hasIPConfigOverride = false
+	if inventory != nil && inventory.SystemVendor != nil && strings.EqualFold(inventory.SystemVendor.Manufacturer, ZVM_VENDOR_ID) &&
+		!strings.HasSuffix(inventory.SystemVendor.ProductName, VM_CTRL_PRG) && (inventory.Boot != nil) {
+		// Commandline for dasd and static IP w/o nmstate might look like:
+		// rd.neednet=1 console=ttysclp0 coreos.live.rootfs_url=http://172.23.236.156:8080/assisted-installer/rootfs.img
+		// ip=10.14.6.3::10.14.6.1:255.255.255.0:master-0.boea3e06.lnxero1.boe:encbdd0:none nameserver=10.14.6.1
+		// ip=[fd00::3]::[fd00::1]:64::encbdd0:none nameserver=[fd00::1] zfcp.allow_lun_scan=0 rd.znet=qeth,0.0.bdd0,0.0.bdd1,0.0.bdd2,layer2=1
+		// rd.dasd=0.0.5235 rd.dasd=0.0.5236 random.trust_cpu=on rd.luks.options=discard ignition.firstboot ignition.platform.id=metal console=tty1 console=ttyS1,115200n8
+		// For nmstate and FCP device the parmline might look like:
+		// rd.neednet=1 console=ttysclp0 coreos.live.rootfs_url=http://172.23.236.156:8080/assisted-installer/rootfs.img zfcp.allow_lun_scan=0
+		// rd.znet=qeth,0.0.bdd0,0.0.bdd1,0.0.bdd2,layer2=1 rd.zfcp=0.0.8004,0x500507630400d1e3,0x4000404800000000
+		// random.trust_cpu=on rd.luks.options=discard ignition.firstboot ignition.platform.id=metal coreos.inst.persistent-kargs="console=tty1 console=ttyS1,115200n8"
+		log.Debugf("Check current boot cmdline: %s\n", strings.TrimSpace(inventory.Boot.CommandLine))
+		// check if config override exists
+		hasIPConfigOverride = strings.Contains(strings.ToLower(strings.TrimSpace(inventory.Boot.CommandLine)), ZVM_IP_CFG_OVER)
+
+		for _, part := range strings.Split(strings.TrimSpace(inventory.Boot.CommandLine), " ") {
+			if strings.HasPrefix(strings.ToLower(part), ZVM_NEEDNET) ||
+				strings.HasPrefix(strings.ToLower(part), ZVM_LUN_SCAN) ||
+				strings.HasPrefix(strings.ToLower(part), ZVM_ZNET) ||
+				strings.HasPrefix(strings.ToLower(part), ZVM_DASD) ||
+				strings.HasPrefix(strings.ToLower(part), ZVM_FCP) ||
+				// For s390x network devices having highly dynamic assigned mac addresses it's necessary to assign a static ip
+				// and nameserver
+				(hasIPConfigOverride && (strings.HasPrefix(strings.ToLower(part), ZVM_IP) ||
+					strings.HasPrefix(strings.ToLower(part), ZVM_NAMESERVER))) {
+				installerArgs = append(installerArgs, "--append-karg", part)
+			}
+		}
+	}
```

Comment:
Please extract all of this code to a named function

Reply:
Moved to named function handling the code.